plugins/connection/zpecloud.py

```diff
@@ -326,23 +326,19 @@ def _wait_job_to_finish(self, job_id: str) -> StringError:
             operation_output_file_url = content.get("output_file", None)
 
             if (
-                operation_status == "Successful"
+                (operation_status == "Successful" or operation_status == "Failed")
                 and operation_output_file_url
                 and len(operation_output_file_url) > 0
             ):
-                self._log_info(f"Job {job_id} finished successfully")
+                self._log_info(f"Job {job_id} finished with status {operation_status}")
```

Comment:
After analyzing I noticed that the status may be something like "Waiting for device", so it is not enough to output the log outside the scope, it would require to check if the status is within "Cancelled", "Timeout", "Successful", "Cancelled". 
So I will leave up to you this change, code looks good the way it is.

Some pseudo-code that may attend my first comment about the log:
```py
if operation_status in ("Cancelled", "Timeout", "Successful", "Cancelled"):
    self._log_info(f"Job {job_id} finished with status {operation_status}")
    if (
        operation_output_file_url
        and len(operation_output_file_url) > 0
    ):
        r = requests.get(operation_output_file_url)
        if isinstance(r.content, bytes):
            msg = r.content.decode("utf-8")
        else:
            msg = r.content

        if operation_status in ("Cancelled", "Timeout"):
            return None, f"Job finished with status {operation_status}. Output: {msg}."
        else:
            return msg, None
    else:
        return (
            None,
            f"Job finished with status {operation_status}. Not output content.",
        )
```
After analyzing I noticed that the status may be something like "Waiting for device", so it is not enough to output the log outside the scope, it would require to check if the status is within those ones from the first line of this pseudo code, so I will leave up to you this change. Code looks good the way it is.

Reply:
Thanks for the analysis.

I will leave it as it is.